manifests/config/apache.pp

```diff
@@ -0,0 +1,284 @@
+  Stdlib::Absolutepath $ssl_cert = $::foreman::server_ssl_cert,
+  Variant[Enum[''], Stdlib::Absolutepath] $ssl_certs_dir = $::foreman::server_ssl_certs_dir,
+  Stdlib::Absolutepath $ssl_key = $::foreman::server_ssl_key,
+  Variant[Enum[''], Stdlib::Absolutepath] $ssl_crl = $::foreman::server_ssl_crl,
+  Optional[String] $ssl_protocol = $::foreman::server_ssl_protocol,
+  String $user = $::foreman::user,
+  Boolean $passenger_prestart = $::foreman::passenger_prestart,
+  Integer[0] $passenger_min_instances = $::foreman::passenger_min_instances,
+  Integer[0] $passenger_start_timeout = $::foreman::passenger_start_timeout,
+  Stdlib::HTTPUrl $foreman_url = $::foreman::foreman_url,
+  Boolean $keepalive = $::foreman::keepalive,
+  Integer[0] $max_keepalive_requests = $::foreman::max_keepalive_requests,
+  Integer[0] $keepalive_timeout = $::foreman::keepalive_timeout,
+  Optional[String] $access_log_format = undef,
+  Boolean $ipa_authentication = $::foreman::ipa_authentication,
+  Optional[Boolean] $selinux = $::foreman::selinux,
+) {
+  $docroot = "${app_root}/public"
+  $suburi_parts = split($foreman_url, '/')
+  $suburi_parts_count = size($suburi_parts) - 1
+  if $suburi_parts_count >= 3 {
+    $suburi_without_slash = join(values_at($suburi_parts, ["3-${suburi_parts_count}"]), '/')
+    if $suburi_without_slash {
+      $suburi = "/${suburi_without_slash}"
+    } else {
+      $suburi = undef
+    }
+  } else {
+    $suburi = undef
+  }
+
+  if $passenger {
+    $passenger_options = {
+      'passenger_app_root' => $app_root,
+      'passenger_min_instances' => $passenger_min_instances,
+      'passenger_start_timeout' => $passenger_start_timeout,
+      'passenger_ruby' => $passenger_ruby,
+    }
+    $passenger_http_prestart = $passenger_prestart ? {
+      true  => "http://${servername}:${server_port}",
+      false => undef,
+    }
+    $passenger_https_prestart = $passenger_prestart ? {
+      true  => "https://${servername}:${server_ssl_port}",
+      false => undef,
+    }
+
+    if $suburi {
+      $custom_fragment = template('foreman/_suburi.conf.erb')
+    } else {
+      $custom_fragment = template('foreman/_assets.conf.erb')
+    }
+
+    $proxy_http_options = {}
+    $proxy_https_options = {}
+
+    if $app_root {
+      file { ["${app_root}/config.ru", "${app_root}/config/environment.rb"]:
+        owner => $user,
+      }
+    }
+  } else {
+    $passenger_options = {}
+    $passenger_http_prestart = undef
+    $passenger_https_prestart = undef
+
+    if $suburi {
+      $custom_fragment = undef
+    } else {
+      $custom_fragment = template('foreman/_assets.conf.erb')
+    }
+
+    $backend = 'http://localhost:3000/'
+
+    $proxy_http_options = {
+      'proxy_preserve_host' => true,
+      'proxy_add_headers'   => true,
+      'request_headers' => [
+        'set X_FORWARDED_PROTO "http"',
+      ],
+      'proxy_pass'          => {
+        'no_proxy_uris' => ['/pulp', '/streamer', '/pub'],
```

Comment:
I want to verify these are actually needed

Reply:
I can't see why they wouldn't be needed since Foreman owns the root URL.